docs/tutorials/cooperative_groups_tutorial.rst

```diff
@@ -0,0 +1,312 @@
+.. code-block:: cpp
+
+  // Number of blocks to launch.
+  constexpr unsigned int num_blocks = 1;
+
+  // Number of threads in each kernel block.
+  constexpr unsigned int threads_per_block = 64;
+
+  // Total element count of the input vector.
+  constexpr unsigned int size = num_blocks * threads_per_block;
+
+  // Total elements count of a tiled_partition.
+  constexpr unsigned int partition_size = 16;
+
+  // Total size (in bytes) of the input vector.
+  constexpr size_t size_bytes = sizeof(unsigned int) * size;
+
+  static_assert(threads_per_block % partition_size == 0,
+                "threads_per_block must be a multiple of partition_size");
+
+3. Allocate and copy input to global memory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: cpp
+
+  // Allocate device memory for the input and output matrices.
+  unsigned int* d_vector{};
+  unsigned int* d_block_reduced{};
+  unsigned int* d_partition_reduced{};
+  HIP_CHECK(hipMalloc(&d_vector, size_bytes));
+  HIP_CHECK(hipMalloc(&d_block_reduced, sizeof(unsigned int) * h_block_reduced.size()));
+  HIP_CHECK(hipMalloc(&d_partition_reduced, sizeof(unsigned int) * h_partition_reduced.size()));
+
+  // Transfer the input vector to the device memory.
+  HIP_CHECK(hipMemcpy(d_vector, h_vector.data(), size_bytes, hipMemcpyHostToDevice));
+
+4. Launch the kernel
+~~~~~~~~~~~~~~~~~~~~
+
+Use the ``hipLaunchCooperativeKernel`` to be able to use cooperative groups.
+
+.. code-block:: cpp
+
+  void* params[] = {&d_vector, &d_block_reduced, &d_partition_reduced};
+  // Launching kernel from host.
+  HIP_CHECK(hipLaunchCooperativeKernel(vector_reduce_kernel<partition_size>,
+                                       dim3(num_blocks),
+                                       dim3(threads_per_block),
+                                       params,
+                                       0,
+                                       hipStreamDefault));\
+
+  // Check if the kernel launch was successful.
+  HIP_CHECK(hipGetLastError());
+
+5. Save the results from global memory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can save the device result to host side ``std::vector`` with ``hipMemcpy()``.
+
+.. code-block:: cpp
+
+  // Transfer the result back to the host.
+  HIP_CHECK(hipMemcpy(h_block_reduced.data(),
+                      d_block_reduced,
+                      sizeof(unsigned int) * h_block_reduced.size(),
+                      hipMemcpyDeviceToHost));
+
+  HIP_CHECK(hipMemcpy(h_partition_reduced.data(),
+                      d_partition_reduced,
+                      sizeof(unsigned int) * h_partition_reduced.size(),
+                      hipMemcpyDeviceToHost));
+
+6. Free the global memory
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Always clean up the global memory at the end of the application.
+
+.. code-block:: cpp
+
+  // Free the resources on the device.
+  HIP_CHECK(hipFree(d_vector));
+  HIP_CHECK(hipFree(d_block_reduced));
+  HIP_CHECK(hipFree(d_partition_reduced));
+
+
+Conclusion
```

Comment:
The Conclusion heading should be H2 (like prerequisites

Reply:
Fixed.